pkg/systemd/services/service-details.jsx

```diff
@@ -319,6 +332,45 @@ export class ServiceDetails extends React.Component {
                 });
     }
 
+    deleteAction() {
+        // Stop timer so we don't get race conditions when the unit is gone.
+        if (this.interval)
+            clearInterval(this.interval);
+
+        const promises = [];
+        if (this.props.unit.UnitFileState === "enabled") {
+            if (this.props.unit.ActiveState === "active" || this.props.unit.ActiveState === "activating")
+                promises.push(this.unitAction("StopUnit"));
+            if (this.props.unit.ActiveState === "failed")
+                promises.push(this.unitAction("ResetFailedUnit", []));
+            promises.push(this.unitFileAction("DisableUnitFiles", undefined));
+        }
+
+        // Remove unit files
+        Promise.all(promises).then(() => {
+            const deletions = [];
+
+            deletions.push(cockpit.file(this.props.unit.FragmentPath, { superuser: "required" }).replace(null));
+            this.props.unit.Triggers.forEach(trigger => {
+                trigger = trigger.replace('.', '_2e');
```

Comment:
Hmm, is that really the only char that systemd escapes? Would IMHO be safer to query this from the timer unit object.

Reply:
Looking at  the `systemd-esacpe` man page, more can be escaped.. Not sure how we can query this from the timer object?

Reply:
Actually systemd-escape does not .. escape the . and the triggers are a list of services so "foo.service", so I assume `hallöchen.service` should be escaped.

Reply:
So hmm as we only have the service name I'm not sure how we can obtain the FragmentPath differently.

Reply:
I played around a little bit with this, but I can't make it work:
```sh
cat <<EOF > /etc/systemd/system/foo.timer
[Timer]
OnCalendar=*:1/2
Unit=$(systemd-escape hallöchen.service)
EOF

cat <<EOF /etc/systemd/system/hallöchen.service
[Service]
ExecStart=/bin/true
EOF
```

With `Unit=hallöchen.service`, systemd complains about "Unit type not valid, ignoring: hallöchen.service", so I escaped it, it ends up as `Unit=hall\xc3\xb6chen.service`. But systemd does not accept that either:

    foo.timer: Refusing to start, unit hall\xc3\xb6chen.service to trigger not loaded

But nevertheless, I figure this will work somehow, and that interaction with shell is somehow broken.

But anyway, the proper way to translate an unit name into an object path is [GetUnit()](https://www.freedesktop.org/wiki/Software/systemd/dbus/):
```
❱❱❱ busctl call org.freedesktop.systemd1 /org/freedesktop/systemd1 org.freedesktop.systemd1.Manager GetUnit s udisks2.service
o "/org/freedesktop/systemd1/unit/udisks2_2eservice"
```
Please use that instead of the manual escaping and hardcoding the `/org/freedesktop/systemd1/unit/...` object path.

Reply:
BTW, just *if* you want to explore something new: This deleteAction() function would be an excellent candidate for `async`/`await`. This should get rid of all the Promise.all(), duplicated .catch(), etc.

Reply:
We could async/await it, but we need to remember that we need to use try/catch then for error handling iirc.